app/Http/Controllers/Store/PaymentController.php

```diff
@@ -23,8 +25,44 @@ public function __construct()
      */
     public function show($paymentUuid)
     {
-        return response($paymentUuid, 200)
-            ->header('Content-Type', 'text/plain');
+        // For later
+        $vouchers = [];
+        $trader = "trader";
+        $number_to_pay = 0;
+
+        // Find the StateToken of a given uuid
+        $state_token = StateToken::where('uuid', $paymentUuid)->first();
+        if ($state_token !== null) {
+
+            // Get the VoucherStates with this StateToken
+            $voucher_states = $state_token
+                ->voucherStates()
+                ->get();
+
+            // Get the voucher codes of states TODO - better
```

Comment:
Solid method for getting the vouchers; Two queries and a bit of collection management :+1:

I think (mildly untested) The "better" solution would be, given `StateToken --< VoucherState >-- Voucher` : 
Add the following function to StateToken, which allows it to reach Vouchers through VoucherStates, like a many-many relationship.

`
   public function vouchers()
    {
        return $this->belongsToMany(Voucher::class, (new VoucherState)->getTable(), 'state_token_id', 'voucher_id');
    }
`
Then do `$vouchers = $state_token->vouchers`